src/vmm/src/devices/virtio/queue.rs

```diff
@@ -495,11 +494,14 @@ impl Queue {
         // We calculate offset into `ring` field.
         let used_ring_offset = std::mem::size_of::<u16>()
             + std::mem::size_of::<u16>()
-            + std::mem::size_of::<UsedElement>() * usize::from(index);
+            + std::mem::size_of::<UsedElement>() * usize::from(index % self.actual_size());
         let used_element_address = self.used_ring.unchecked_add(usize_to_u64(used_ring_offset));
 
-        mem.write_obj(used_element, used_element_address)
-            .map_err(QueueError::UsedRing)
+        // SAFETY:
+        // `used_element_address` param is bounded by size of the queue as `index` is
+        // modded by `actual_size()`.
+        mem.write_obj(used_element, used_element_address).unwrap();
+        Ok(())
```

Comment:
`write_obj` returns an error when the object your are trying to write into the address does not fit in the available memory, so the `SAFETY` comment above here is not accurate; you might have an address pointing inside the used ring but if you try to write something that does not fit in the ring this will fail.

Now, this thing will not happen here because you are actually writing a `UsedElement`. But I still don't understand why we should change the way of handling errors. Why is this needed for your PR (or the subsequent one)? I think it is not.

Reply:
Removed commit from this PR.